lib/maplibre-compose/src/commonMain/kotlin/dev/sargunv/maplibrecompose/compose/source/rememberGeoJsonSource.kt

```diff
@@ -4,30 +4,40 @@ import androidx.compose.runtime.Composable
 import androidx.compose.runtime.key
 import dev.sargunv.maplibrecompose.core.source.GeoJsonOptions
 import dev.sargunv.maplibrecompose.core.source.GeoJsonSource
-import dev.sargunv.maplibrecompose.core.source.Source
 import io.github.dellisd.spatialk.geojson.GeoJson
 
+/**
+ * Remember a new [GeoJsonSource] with the given [id] and [options] from the GeoJson data at the
+ * given [uri].
+ *
+ * @throws IllegalArgumentException if a layer with the given [id] already exists.
+ */
 @Composable
 @Suppress("NOTHING_TO_INLINE")
 public inline fun rememberGeoJsonSource(
   id: String,
   uri: String,
   options: GeoJsonOptions = GeoJsonOptions(),
-): Source =
+): GeoJsonSource =
```

Comment:
Okay to merge for now but I think we'll probably want to split these types into mutable and non mutable interfaces, and return the non mutable interface here so there's only one path for setting the uri (via the Composable parameter). 

Making mutations internal would also work, but I'd like to split the modules between core and compose so internal wouldn't work forever

Reply:
Why should Source(s) be mutable at all?

Reply:
The geojson data can be updated. We use that in the demo app right now on the camera follow demo, to move the location puck.

In compose that's done with a parameter to the Composable, but the underlying non-compose model just has a setter for it.

Reply:
Oh, right! I didn't look at the code, I wrongly assumed that (in compose) the whole GeoJsonSource would be replaced on any change to it.